File: ECC.py
from tinyec import registry
import numpy as np
from os.path import exists


class ECC_Instance(object):
    """docstring for ."""

    def __init__(self, C, name):
        self.C = C
        self.name = name
        self.load_public_key()

    def generate_private_key(self):
        self.priv_key = np.random.randint(1,100)
        with open(f'data/priv_key_{self.name}.txt', "w") as priv_file:
            priv_file.write(f"{self.priv_key}")

    def load_public_key(self):
        if exists(f'data/priv_key_{self.name}.txt'):
            with open(f'data/priv_key_{self.name}.txt', 'r') as f:
                self.priv_key = int(f.read())
        else:
            self.generate_private_key()
        self.pub_key = self.priv_key * self.C.g

    def generate_shared_key(self,other_pub_key):
        self.shared_key = self.priv_key*other_pub_key

    def get_public_key(self):
        return self.pub_key

    def get_shared_key(self):
        return self.shared_key

File: ECC.ts
import * as ecc_math from "simple-js-ec-math";
import Cookies from "js-cookie";

export class ECCInstance {
  C: ecc_math.Curve;
  G: ecc_math.ModPoint;
  name: string;
  privateKey: number;
  publicKey: string;
  sharedKey: ecc_math.ModPoint;

  constructor(C: ecc_math.Curve, G: ecc_math.ModPoint, name: string) {
    this.C = C;
    this.G = G;
    this.name = name;
    this.loadPublicKey();
  }

  generatePrivateKey(): void {
    let min = 1;
    let max = 1000;

    this.privateKey = Math.floor(Math.random() * (max - min + 1) + min);

    //ALTERNATE OPTION
    // localStorage.setItem("lastname", "Smith");
    // console.log(localStorage.getItem("lastname"));
    if (Cookies.get("privateKey_" + this.name) == undefined) {
      Cookies.set("privateKey_" + this.name, this.privateKey, { expires: 365 });
    }
  }

  loadPublicKey(): void {
    if (Cookies.get("privateKey_" + this.name) == undefined) {
      this.generatePrivateKey();
    } else {
      this.privateKey =+ Cookies.get("privateKey_" + this.name);
    }
    this.publicKey = this.C.multiply(this.G, this.privateKey);
  }

  generateSharedKey(publicKey: string): void {
    this.sharedKey = this.C.multiply(publicKey, this.privateKey);
  }

  getPublicKey(): string {
    return this.publicKey;
  }

  getSharedKey(): string {
    return this.sharedKey;
  }

  getName() {
    return this.name;
  }

  getPrivateKey() {
    return this.privateKey;
  }

  clearKeys() {
    Cookies.remove("privateKey_" + this.name);
  }
}
